### ISSUE TITLE
Document current usage recommendations

### ISSUE THREAD

**Issue by garvankeeley** [mozilla-mobile/telemetry-ios#21]
1) is not thread-safe, call only from main
2) is only for CorePing and UI events which are inherently rate-limited by user interaction.

**Comment by Sdaswani** [comment]
Per 1, can we also clarify that the call is non-blocking, so calling from main is relatively safe?

**Comment by garvankeeley** [comment]
Will do, you are correct, all calls are main-thread async. 
They can result in main thread contention if called at excessive rate.

**Comment by Sdaswani** [comment]
Yeah so we need to change that down the line. It's not good API behavior to expect clients to build their own rate limiting.

**Comment by garvankeeley** [comment]
See also https://github.com/mozilla-mobile/telemetry-ios/issues/12
